src/sql.rs

```diff
@@ -523,11 +527,23 @@ impl Sql {
                 .await?;
         }
 
+        let mut lock = self.config_cache.write().await;
+        lock.insert(key.to_string(), value.map(|s| s.to_string()));
+        drop(lock);
+
         Ok(())
     }
 
     /// Get configuration options from the database.
     pub async fn get_raw_config(&self, key: impl AsRef<str>) -> Result<Option<String>> {
```

Comment:
why not also update cache on reading a value? so, before `Ok(value)` below?

i think, this is how caches are usually populated.